Sources/Parley/Views/ParleyMessageView/ParleyMessageView.swift

```diff
@@ -226,22 +238,14 @@ final class ParleyMessageView: UIView {
                 renderGradients()
             }
             
-            if let media = message.media, let mediaIdUrl = URL(string: media.id) {
-                let url = mediaIdUrl.pathComponents.dropFirst().dropFirst().joined(separator: "/")
-                findImageRequest = messageRepository.find(media: url, onSuccess: onFindSuccess(image:), onFailure: onFindError(error:))
-            } else if let id = message.id {
-                findImageRequest = messageRepository.findImage(id, onSuccess: onFindSuccess(image:), onFailure: onFindError(error:))
-            } else {
-                print("Failed to render image for message \(message.id)")
+            Parley.shared.imageRepository.getRemoteImage(for: mediaId) { result in
+                switch result {
+                case .success(let displayModel):
+                    onFindSuccess(image: displayModel.image)
+                case .failure(let error):
+                    onFindError(error: error)
+                }
```

Comment:
- // Move existing `cancel()` to top
- Remove cancelable since doesn't seem needed anymore
- In onFindSuccess check if message is still the same, else ignore (next time, imageRepo will load it)
- Also in onFindError
- Document that check/guard